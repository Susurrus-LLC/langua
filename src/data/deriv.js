export const defData = {
  words: 5,
  type: 'any',
  lexemes: [
    {
      lexeme: 'nzi',
      definition: 'tree'
    },
    {
      lexeme: 'ilu',
      definition: 'flower'
    },
    {
      lexeme: 'ru',
      definition: 'person'
    },
    {
      lexeme: 'kwərə',
      definition: 'goat'
    },
    {
      lexeme: 'lɛtsi',
      definition: 'sword'
    }
  ],
  derivations: [
    {
      derivation: '-vo',
      gloss: 'person',
      definition:
        'Creates a noun referring to a person associated with the base.'
    },
    {
      derivation: '-lu',
      gloss: 'place',
      definition: 'Creates a noun denoting the location of the base.'
    },
    {
      derivation: 'ña-',
      gloss: 'abstract',
      definition: 'Creates an abstract noun.'
    },
    {
      derivation: 'k-sa',
      gloss: 'collective',
      definition: 'Creates a noun referring to a collection of bases.'
    }
  ],
  results: undefined
}
