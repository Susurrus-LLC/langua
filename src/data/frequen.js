export const defData = {
  corpus:
    'Nzɔ de’áxa Tsənaxa, ño lo sɛtsa’ɔk, ño lo de’áxa u Kasaga, ño lo mɛya ayru Tol o Əbe kwə, sa nzɔ xa mpaskale: Steyə rabɛ lu ñire lu mvunə poxe o lu pɛlɔ ah lo mɛwɛ axe, maxə ɔra latsi əbɔwə səyo.',
  consonants:
    'b/d/f/g/k/kw/l/m/mp/mv/n/nt/nts/nz/ñ/ŋk/ŋkw/p/r/s/t/ts/v/w/x,h/y/’',
  vowels: 'a,á/ɛ,ɛ́/e,é/ə,ə́/i,í/ɔ,ɔ́/o,ó/u,ú',
  distinguishCase: false,
  analyzed: false,
  results: undefined,
  hovered: undefined,
  filterSeg: 'all',
  filterSyll: 'all',
  filterWords: 'all'
}
