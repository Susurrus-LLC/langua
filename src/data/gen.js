export const defData = {
  subpatterns: [
    {
      selected: 'V',
      subpattern: 'a/e/i/o/u'
    },
    {
      selected: 'C',
      subpattern: 'p/t/k/b/d/g/s/m/n/l/r'
    },
    {
      selected: 'N',
      subpattern: 'm/n'
    }
  ],
  pattern: 'V/CV/CVN/CVCV/CVCVN',
  words: 100,
  newline: false,
  filterdupes: false
}

export const vars = [
  'A',
  'B',
  'C',
  'D',
  'E',
  'F',
  'G',
  'H',
  'I',
  'J',
  'K',
  'L',
  'M',
  'N',
  'O',
  'P',
  'Q',
  'R',
  'S',
  'T',
  'U',
  'V',
  'W',
  'X',
  'Y',
  'Z'
]
