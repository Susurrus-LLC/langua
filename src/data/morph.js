export const defData = {
  categories: [
    'V=ieaou',
    'L=īēāōū',
    'C=ptcqbdgmnlrhs',
    'F=ie',
    'B=ou',
    'S=ptc',
    'Z=bdg'
  ],
  rewriteRules: ['lh=lj'],
  soundChanges: [
    '[sm]//_#',
    'i/j/_V',
    'L/V/_',
    'e//Vr_#',
    'v//V_V',
    'u/o/_#',
    'gn/nh/_',
    'S/Z/V_V',
    'c/i/F_t',
    'c/u/B_t',
    'p//V_t',
    'ii/i/_',
    'e//C_rV'
  ],
  lexicon: [
    'lector',
    'doctor',
    'focus',
    'jocus',
    'districtus',
    'cīvitatem',
    'adoptare',
    'opera',
    'secundus',
    'fīliam',
    'pōntem'
  ],
  outputFormat: 'oo',
  showDiff: true,
  showChanges: true,
  ruleReport: true,
  rewriteOutput: true,
  results: undefined
}
