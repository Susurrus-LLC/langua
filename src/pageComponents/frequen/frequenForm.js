import React from 'react'
import PropTypes from 'prop-types'

import Form from '../../components/form'

import sharedStyles from '../../components/form/sharedForm.module.sass'

const FrequenForm = () => <Form name='frequen-form' />

FrequenForm.propTypes = {
  styles: PropTypes.string
}

export default FrequenForm
