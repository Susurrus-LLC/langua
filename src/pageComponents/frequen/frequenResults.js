import React from 'react'
import classNames from 'classnames'
import {
  FlexibleWidthXYPlot,
  HorizontalBarSeries,
  HorizontalGridLines,
  VerticalGridLines,
  XAxis,
  YAxis,
  Hint
} from 'react-vis'
import 'react-vis/dist/style.css'
import PropTypes from 'prop-types'

import Results from '../../components/results'
import Control from '../../components/control'
import ControlPiece from '../../components/control/piece'

import sharedStyles from '../../components/results/sharedResults.module.sass'

const FrequenResults = ({
  styles,
  results,
  hovered,
  filterSeg,
  analyzed,
  onChangeInput,
  onMouseOver,
  onMouseOut
}) => {
  const twoDecimals = num => (Math.round(num * 100) / 100).toFixed(2)

  const whichData = () => {
    switch (filterSeg) {
      case 'consonants':
        return results.consonants
      case 'vowels':
        return results.vowels
      default:
        return results.all
    }
  }

  const headerCells = () => {
    let data = []

    for (let prop in results) {
      if (results.hasOwnProperty(prop)) {
        let count = 0

        results[prop].forEach(el => {
          count += el.count
        })

        data.push({
          name: prop,
          total: count
        })
      }
    }

    const filterData = () => {
      let newData = {}

      data.forEach(el => {
        if (el.name === filterSeg) {
          newData = {
            name: el.name,
            total: el.total
          }
        }
      })

      return newData
    }

    if (filterSeg !== 'all') {
      return (
        <th className={sharedStyles.headerCell}>
          {`% of ${filterData().name}`}
          <br />
          {`total: ${filterData().total}`}
        </th>
      )
    } else {
      return data.map((d, i) => (
        <th className={sharedStyles.headerCell} key={i}>
          {`% of ${d.name}`}
          <br />
          {`total: ${d.total}`}
        </th>
      ))
    }
  }

  const dataRows = () => {
    const allData = JSON.parse(JSON.stringify(whichData())).sort((a, b) => {
      return b.count - a.count
    })

    const findPercent = (arr, y) => {
      let i = -1

      for (let j = 0; j < arr.length; j++) {
        if (arr[j].y === y) {
          i = j
        }
      }

      if (i > -1) {
        return `${twoDecimals(arr[i].x)}%`
      } else {
        return `${twoDecimals(0)}%`
      }
    }

    return allData.map(seg => {
      const segment = seg.y
      const count = seg.count
      const percent = twoDecimals(seg.x)
      const consPercent = findPercent(results.consonants, segment)
      const vowsPercent = findPercent(results.vowels, segment)

      return (
        <tr className={sharedStyles.dataRow} key={seg.i}>
          <td className={sharedStyles.dataCell}>{segment}</td>
          <td className={sharedStyles.dataCell}>{count}</td>
          <td className={sharedStyles.dataCell}>{`${percent}%`}</td>
          {filterSeg === 'all' ? (
            <>
              <td className={sharedStyles.dataCell}>
                {seg.type === 'consonant' ? consPercent : null}
              </td>
              <td className={sharedStyles.dataCell}>
                {seg.type === 'vowel' ? vowsPercent : null}
              </td>
            </>
          ) : null}
        </tr>
      )
    })
  }

  const axisStyle = {
    fontSize: `0.75rem`,
    text: {
      fill: `#777`
    },
    line: {
      stroke: `#aaa`
    }
  }

  const gridStyle = {
    stroke: `#aaa`
  }

  if (analyzed) {
    return (
      <Results>
        <Control>
          <ControlPiece>
            <select
              id='filterSeg'
              name='filterSeg'
              value={filterSeg}
              onChange={onChangeInput}
            >
              <option value='all'>All Segments</option>
              <option value='consonants'>Consonants</option>
              <option value='vowels'>Vowels</option>
            </select>
          </ControlPiece>
        </Control>
        <div
          className={classNames(
            sharedStyles.barChart,
            sharedStyles.resultsSection
          )}
        >
          <FlexibleWidthXYPlot
            yType='ordinal'
            height={
              v.ms1 * 16 * whichData().length /* bar height */ +
              v.ms3 * (whichData().length - 1) /* bar spacing */ +
              64 /* top and bottom axes */
            }
            margin={`${v.ms3 * 16} ${v.ms5 * 16}`}
            onMouseLeave={onMouseOut}
          >
            <VerticalGridLines style={gridStyle} />
            <HorizontalGridLines style={gridStyle} />
            <XAxis
              tickFormat={t => `${t}%`}
              style={axisStyle}
              orientation='top'
            />
            <XAxis
              tickFormat={t => `${t}%`}
              style={axisStyle}
              orientation='bottom'
            />
            <YAxis style={axisStyle} orientation='left' />
            <YAxis style={axisStyle} orientation='right' />
            <HorizontalBarSeries
              data={whichData()}
              animation
              colorType='literal'
              onValueMouseOver={onMouseOver}
              onMouseOut={onMouseOut}
            />
            {hovered ? (
              <Hint
                value={{
                  [`/${hovered.y}/ (${hovered.type})`]: `${
                    hovered.count
                  } (${twoDecimals(hovered.x)}%)`
                }}
              />
            ) : null}
          </FlexibleWidthXYPlot>
        </div>
        <div
          className={classNames(
            sharedStyles.dataTable,
            sharedStyles.resultsSection
          )}
        >
          <table className={sharedStyles.table}>
            <thead className={sharedStyles.tableHead}>
              <tr className={sharedStyles.headerRow}>
                <th className={sharedStyles.headerCell}>Segment</th>
                <th className={sharedStyles.headerCell}>Count</th>
                {headerCells()}
              </tr>
            </thead>
            <tbody className={sharedStyles.tableBody}>{dataRows()}</tbody>
          </table>
        </div>
      </Results>
    )
  } else {
    return <div className={sharedStyles.results} />
  }
}

FrequenResults.propTypes = {
  styles: PropTypes.object,
  results: PropTypes.shape({
    all: PropTypes.arrayOf(
      PropTypes.shape({
        color: PropTypes.string.isRequired,
        count: PropTypes.number.isRequired,
        i: PropTypes.number.isRequired,
        type: PropTypes.string.isRequired,
        x: PropTypes.number.isRequired,
        y: PropTypes.string.isRequired
      })
    ),
    consonants: PropTypes.arrayOf(
      PropTypes.shape({
        color: PropTypes.string.isRequired,
        count: PropTypes.number.isRequired,
        i: PropTypes.number.isRequired,
        type: PropTypes.string.isRequired,
        x: PropTypes.number.isRequired,
        y: PropTypes.string.isRequired
      })
    ),
    vowels: PropTypes.arrayOf(
      PropTypes.shape({
        color: PropTypes.string.isRequired,
        count: PropTypes.number.isRequired,
        i: PropTypes.number.isRequired,
        type: PropTypes.string.isRequired,
        x: PropTypes.number.isRequired,
        y: PropTypes.string.isRequired
      })
    )
  }),
  hovered: PropTypes.shape({
    color: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
    i: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    x: PropTypes.number.isRequired,
    y: PropTypes.string.isRequired
  }),
  filterSeg: PropTypes.oneOf(['consonants', 'vowels', 'all']).isRequired,
  analyzed: PropTypes.bool.isRequired,
  onChangeInput: PropTypes.func.isRequired,
  onMouseOver: PropTypes.func.isRequired,
  onMouseOut: PropTypes.func.isRequired
}

export default FrequenResults
