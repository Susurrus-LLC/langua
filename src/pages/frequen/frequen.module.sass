@import ../../styles/variables.module

.input
  width: 100%

.corpus-input
  height: $ms6 * 2rem
