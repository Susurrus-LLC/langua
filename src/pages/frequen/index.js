import React from 'react'
import { graphql, useStaticQuery } from 'gatsby'

import Layout from '../../components/layout'
import SEO from '../../components/seo'
import Notice from '../../components/notice'
import Tool from '../../components/tool'

import FrequenForm from '../../pageComponents/frequen/frequenForm'
import FrequenResults from '../../pageComponents/frequen/frequenResults'
import FrequenService from '../../services/frequenService'

import styles from './frequen.module.sass'

const FrequenPage = () => {
  const data = useStaticQuery(graphql`
    query FrequenToolQuery {
      site {
        siteMetadata {
          title
          toolInfo {
            frequen {
              title
              link
              description
            }
          }
        }
      }
    }
  `)

  const metadata = data.site.siteMetadata
  const toolInfo = metadata.toolInfo.frequen

  return (
    <Layout>
      <SEO
        title={`${metadata.title}${toolInfo.title}`}
        canonical={toolInfo.link}
      />
      <Tool toolInfo={toolInfo}>
        <Notice>This tool is still in development.</Notice>
        <FrequenForm />
        <FrequenResults />
      </Tool>
    </Layout>
  )
}

export default FrequenPage
