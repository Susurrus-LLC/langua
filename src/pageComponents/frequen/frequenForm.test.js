import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import FrequenForm from './frequenForm'
import { defData } from '../../data/frequen'

describe('FrequenForm', () => {
  const styles = {
    styles: 'styles',
    corpusInput: 'corpusInput',
    input: 'input',
    consonants: 'consonants',
    vowels: 'vowels'
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <FrequenForm
        styles={styles}
        corpus={defData.corpus}
        consonants={defData.consonants}
        vowels={defData.vowels}
        distinguishCase
        onChangeInput={jest.fn()}
        onAnalyze={jest.fn()}
      />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately', () => {
    let tree = renderer
      .create(
        <FrequenForm
          styles={styles}
          corpus={defData.corpus}
          consonants={defData.consonants}
          vowels={defData.vowels}
          distinguishCase
          onChangeInput={jest.fn()}
          onAnalyze={jest.fn()}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
