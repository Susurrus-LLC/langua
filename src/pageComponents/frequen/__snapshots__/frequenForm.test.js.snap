// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrequenForm renders accurately 1`] = `
<form
  className="form"
  name="frequen-form"
>
  <div
    className="part"
  >
    <h5
      className="sectionTitle"
    >
      Text Corpus
    </h5>
    <textarea
      aria-label="Text corpus"
      className="corpusInput input"
      id="corpus"
      name="corpus"
      onChange={[MockFunction]}
      value="Nzɔ de’áxa Tsənaxa, ño lo sɛtsa’ɔk, ño lo de’áxa u Kasaga, ño lo mɛya ayru Tol o Əbe kwə, sa nzɔ xa mpaskale: Steyə rabɛ lu ñire lu mvunə poxe o lu pɛlɔ ah lo mɛwɛ axe, maxə ɔra latsi əbɔwə səyo."
    />
  </div>
  <div
    className="consonants part"
  >
    <h5
      className="sectionTitle"
    >
      Consonants
    </h5>
    <input
      aria-label="Consonants"
      className="input"
      id="consonants"
      name="consonants"
      onChange={[MockFunction]}
      type="text"
      value="b/d/f/g/k/kw/l/m/mp/mv/n/nt/nts/nz/ñ/ŋk/ŋkw/p/r/s/t/ts/v/w/x,h/y/’"
    />
  </div>
  <div
    className="vowels part"
  >
    <h5
      className="sectionTitle"
    >
      Vowels
    </h5>
    <input
      aria-label="Vowels"
      className="input"
      id="vowels"
      name="vowels"
      onChange={[MockFunction]}
      type="text"
      value="a,á/ɛ,ɛ́/e,é/ə,ə́/i,í/ɔ,ɔ́/o,ó/u,ú"
    />
  </div>
  <div
    className="control part"
  >
    <div
      className="controlSide controlLeft"
    >
      <div
        className="controlPiece"
      >
        <button
          className="btn neutral"
          onClick={[MockFunction]}
          role="button"
          type="submit"
        >
          Analyze
        </button>
      </div>
      <div
        className="controlPiece"
      >
        <label
          htmlFor="distinguishCase"
        >
          <input
            checked={true}
            id="distinguishCase"
            name="distinguishCase"
            onChange={[MockFunction]}
            type="checkbox"
            value="distinguishCase"
          />
           
          distinguish upper and lower case
        </label>
      </div>
    </div>
    <div
      className="controlSide controlRight"
    >
      <div
        className="controlPiece"
      >
        <button
          className="btn success"
          id="save"
          onClick={[MockFunction]}
          role="button"
          type="button"
        >
          Save
        </button>
      </div>
      <div
        className="controlPiece"
      >
        <button
          className="btn"
          id="open"
          onClick={[Function]}
          role="button"
          type="button"
        >
          Open
        </button>
        <input
          className="hidden"
          id="file"
          name="file"
          onChange={[MockFunction]}
          type="file"
        />
      </div>
    </div>
  </div>
</form>
`;
